### ISSUE TITLE
Storybook 7 development build won't start with React/Webpack

### ISSUE THREAD

**Issue by lunaris** [nrwl/nx#18295]
### Current Behavior

We've implemented our UI kit as an Nx library (`libs/ui`) written using TypeScript and React. In case it's relevant we are making use of Babel and Styled Components also. We recently upgraded to 16.5.3 and as part of that, bumped to Storybook 7. Prior to this, development builds of the library's Storybook (`nx storybook ui`) worked fine, as did production builds (`nx build-storybook ui`). Now that we have upgraded, development builds crash with no error output. Production builds still seem to work fine.

### Expected Behavior

`nx storybook ui` should start a development server as it always has.

### Steps to Reproduce

I can reproduce this issue on a Macbook M1 (Arm64), running inside a VSCode devcontainer running Linux. Steps to reproduce:

* Generate fresh Storybook configuration e.g. `nx g @nx/storybook:configuration ui --tsConfiguration=true`
* Select the React/Webpack framework option
* Attempt to run the development build with `nx storybook ui`
* Observe a crash with no error message that appears to be linked to `@nx/react/plugins/storybook`

<img width="940" alt="CleanShot 2023-07-25 at 10 17 10@2x" src="https://github.com/nrwl/nx/assets/163793/ff305147-c3e9-4f9d-92b6-8354e9d644f4">

* No errors are produced by setting the Storybook executor's `loglevel` to `silly` (or any lesser log level like `verbose`), nor by passing `verbose` to the Nx command.

### Nx Report

```shell
Node   : 16.19.1
   OS     : linux-arm64
   pnpm   : 8.6.2
   
   nx                 : 16.5.3
   @nx/js             : 16.5.3
   @nx/jest           : 16.5.3
   @nx/linter         : 16.5.3
   @nx/workspace      : 16.5.3
   @nx/cypress        : 16.5.3
   @nx/devkit         : 16.5.3
   @nx/eslint-plugin  : 16.5.3
   @nx/express        : 16.5.3
   @nx/next           : 16.5.3
   @nx/node           : 16.5.3
   @nx/react          : 16.5.3
   @nx/storybook      : 16.5.3
   @nrwl/tao          : 16.5.3
   @nx/web            : 16.5.3
   @nx/webpack        : 16.5.3
   nx-cloud           : 16.1.0
   typescript         : 5.1.6
```

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

This issue doesn't appear on a Linux/AMD x86/64 machine/architecture. I've hacked and console.logged my way through `node_modules` and I think the issue is occurring in `nx/src/native/index.js`'s `isMusl` function. For some reason, when Nx boots up, this function runs fine, but when it is called as part of the Storybook devserver (seemingly triggered by the React plugin's building of `createProjectGraphAsync`), it crashes. In doing so it produces no error logs. I think I've pinned it down to `isMusl`'s use of `process.report.getReport`, which seems to just terminate the process. Wrapping it in a `try`/`catch` doesn't produce any exception -- it just hard crashes. The only workaround I've managed to get working is just to avoid calling it and use the fallback code that checks `ldd` -- this appears to work fine and doing so allows the Storybook server to boot up. Searching the web I can't find anything that would indicate why `process.report.getReport` might be failing in this way -- could it be some shim or fill that Storybook is inserting that behaves differently? Things I've tried:

* Printing `process.report.getReport.toString()` -- works fine and the code displayed is the same in the initial bootup of Nx (where the call succeeds) and in the offending call (where a silent crash happens)
* Passing an explicit error argument to `getReport` -- doesn't change the behaviour
* Storing `process.report.getReport` in a variable to see if on multiple visits the value is changing (e.g. due to a polyfill) -- does not appear to be the case.

For now the best workaround seems to be avoiding calling it. I tried using the React/Vite framework for the Storybook but in doing so suffered other issues (where `process` was undefined). All thoughts and help welcome, and thanks for the great work on Nx!

**Comment by justingorham-tesouro** [comment]
We are in a similar situation. We attempted upgrading from Nx 16.3 to 16.5. The first runtime issue was related to `postcss`. We had to download that as a devDependency. When we try to run storybook after the migration, the following error is pretty consistent.
```
Duplicate __self prop found. You are most likely using the deprecated transform-react-jsx-self Babel plugin. Both __source and __self are automatically set when using the automatic runtime. Please remove transform-react-jsx-source and transform-react-jsx-self from your Babel config.
```

We are using MUI (built on top of emotion)

**Comment by Itrulia** [comment]
@justingorham-tesouro we are facing the same issue with MUI: https://github.com/wepublish/wepublish/pull/1001

I tried a few things (like adding the transformers, updating all babel dependencies, removing transformers from the default babel config etc.)

**Comment by mandarini** [comment]
Hi all! Thanks for filing this issue.

@lunaris I unfortunately cannot reproduce your issue in a new repository. Can you please provide a minimum reproduction?

@justingorham-tesouro @Itrulia I tried out [wepublish/wepublish](https://github.com/wepublish/wepublish) locally, and upgraded, and indeed I run into the `Duplicate __self prop` issue. I have not managed to work around it.

Can you please provide a minimum reproduction @justingorham-tesouro ? The `wepublish` repo is a bit big, and I am not sure I can narrow down what the issue may be. In any case, I tried running Storybook using the `storybook` CLI (navigate to your project's folder and run `npx storybook dev`) and I still could not get it to run (got different errors), so it makes me think that maybe that's not an issue with nx?

**Comment by mandarini** [comment]
Wait I found the issue and I am not sure it's Nx's fault.

**Comment by Itrulia** [comment]
@mandarini that change unfortunately didn't solve it as `process.env.NODE_ENV` was still "development".
I added a console.log in the preset and got:

```
console.log("IS_PRODUCTION", isProduction, isCallerDevelopment, isLoadIntentDevelopment, process.env.NODE_ENV);
```
> `IS_PRODUCTION false undefined true, "development"`

I made sure to check that the env method is being called by storybook.


**BUT**, if I start storybook with the NODE_ENV set to production it works. So this unblocks me, thank you very much!
I will try to make a small reproduction and will check with storybook/next as it seems to be an error on their side.
